src/weather_model_graphs/networkx_utils.py

```diff
@@ -98,40 +98,29 @@ def split_graph_by_edge_attribute(graph, attr):
             f"No subgraphs were created. Check the edge attribute '{attr}'."
         )
 
-    # copy node attributes
-    for subgraph in subgraphs.values():
-        for node in subgraph.nodes:
-            subgraph.nodes[node].update(graph.nodes[node])
-
-    # check that at least one subgraph was created
-    if len(subgraphs) == 0:
-        raise ValueError(
-            f"No subgraphs were created. Check the edge attribute '{attr}'."
-        )
+    return subgraphs
 
-    # copy node attributes
-    for subgraph in subgraphs.values():
-        for node in subgraph.nodes:
-            subgraph.nodes[node].update(graph.nodes[node])
 
-    # check that at least one subgraph was created
-    if len(subgraphs) == 0:
-        raise ValueError(
-            f"No subgraphs were created. Check the edge attribute '{attr}'."
-        )
+def sort_nodes_in_graph(graph):
+    """
+    Creates a new networkx.DiGraph that is a copy of input, but with nodes
+    sorted according to their id
 
-    # copy node attributes
-    for subgraph in subgraphs.values():
-        for node in subgraph.nodes:
-            subgraph.nodes[node].update(graph.nodes[node])
+    Parameters
+    ----------
+    graph : networkx.DiGraph
+        Graph to sort nodes from
 
-    # check that at least one subgraph was created
-    if len(subgraphs) == 0:
-        raise ValueError(
-            f"No subgraphs were created. Check the edge attribute '{attr}'."
-        )
+    Returns
+    -------
+    networkx.DiGraph
+        Graph with sorted nodes
+    """
+    sorted_graph = networkx.DiGraph()
+    sorted_graph.add_nodes_from(sorted(graph.nodes(data=True)))
+    sorted_graph.add_edges_from(graph.edges(data=True))
 
-    return subgraphs
+    return sorted_graph
 
 
 def replace_node_labels_with_unique_ids(graph):
```

Comment:
I am realising now that this is a poor name, maybe I should have called it `replace_node_labels_with_unique_integer_values`

Reply:
Agreed, but maybe not for this PR, as it is orthogonal.